components/button/button-split.js

```diff
@@ -0,0 +1,174 @@
+import './button.js';
+import '../colors/colors.js';
+import '../dropdown/dropdown.js';
+import '../dropdown/dropdown-menu.js';
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+/**
+ * A split button that provides a main action button and slot for a menu.
+ */
+class ButtonSplit extends FocusMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * @ignore
+			 */
+			// eslint-disable-next-line lit/no-native-attributes
+			autofocus: { type: Boolean, reflect: true },
+			/**
+			 * ACCESSIBILITY: A description to be added to the main action button for accessibility when text does not provide enough context
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * Disables the main action and dropdown opener buttons
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * Tooltip text when disabled
+			 * @type {string}
+			 */
+			disabledTooltip: { type: String, attribute: 'disabled-tooltip' },
+			/**
+			 * @ignore
+			 */
+			form: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formaction: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formenctype: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formmethod: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formnovalidate: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formtarget: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			name: { type: String, reflect: true },
+			/**
+			 * Styles the buttons as a primary buttons
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean, reflect: true },
+			/**
+			 * ACCESSIBILITY: REQUIRED: Accessible text for the main action button
+			 * @type {string}
+			 */
+			text: { type: String, reflect: true, required: true },
+			/**
+			 * @ignore
+			 */
+			type: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.container {
```

Comment:
Is the container needed or could these styles just move to the host?

Reply:
So technically they could be moved to the host, but if the consumer changes the display property, it will mess up the internal layout. For this reason, the internal layout is isolated from the host element.

Reply:
Cool yeah that makes sense!